File: pset2/readability/readability.c
#include <cs50.h>
#include <stdio.h>
#include <string.h>
#include <ctype.h>
#include <math.h>


int main(void)
{

    //index = 0.0588 * L - 0.296 * S - 15.8

    string s = get_string("Text: ");

    //average numbers of letters per 100 words: L
    int n = strlen(s);
    int space = 1;
    int letter = 0;
    int punct = 0;

    for (int i = 0; i < n; i++)
    {

        //number of letters
        if ((s[i] >= 'a' && s[i] <= 'z') || (s[i] >= 'A' && s[i] <= 'Z'))
        {
            letter = letter + 1;
        }

        //number of words
        else if (isspace(s[i]))
        {
            space = space + 1;
        }

        else if ((s[i]) == '!' || s[i] == '.' || s[i] == '?')
        {
            punct = punct + 1;

        }
    }



    //calculate L
    float L = 100 * ((float) letter / (float) space);



    //average number of sentences per 100 words in the text: S
    float S = ((float) punct / (float) space) * 100;

    printf("%i\n", letter);
    printf("%i\n", space);
    printf("%i\n", punct);
    printf("%f\n", L);
    printf("%f", S);
    

    //calculate readability
    int readability = round(.0588 * L - 0.296 * S - 15.8);

    if (readability >= 16)
    {

        printf("Grade 16+\n");
    }
    else if (readability < 1)
    {

        printf("Before Grade 1\n");
    }
    else
    {

        printf("Grade %i\n", readability);
    }
}

File: pset6/readability.py
from cs50 import get_string
from sys import argv, exit


s = get_string("Text: ");


n = len(s);
space = 1;
letter = 0;
punct = 0;

for i in range(n):

    if s[i].isalnum()==True:
        letter += 1

    if s[i].isspace()==True:
        space += 1

    if s[i]=="?" or s[i]=="." or s[i]=="!":
        punct += 1

    i += 1


#calculate L and S
L = 100 * (letter / space);
S = (punct / space) * 100


readability = round(.0588 * L - 0.296 * S - 15.8)

if readability >= 16:
    print(f"Grade 16+")

elif readability < 1:
    print(f"Before Grade 1")

else:
    print(f"Grade", readability)
